extensions/elytron-security-common/runtime/src/main/java/io/quarkus/elytron/security/common/runtime/BcryptUtil.java

```diff
@@ -0,0 +1,90 @@
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+import org.wildfly.security.password.interfaces.BCryptPassword;
+import org.wildfly.security.password.spec.EncryptablePasswordSpec;
+import org.wildfly.security.password.spec.IteratedSaltedPasswordAlgorithmSpec;
+import org.wildfly.security.password.util.ModularCrypt;
+
+/**
+ * Utility class used to produce bcrypt hashes using the Modular Crypt Format.
+ */
+public class BcryptUtil {
+
+    private static final SecureRandom random = new SecureRandom();
+    private static final WildFlyElytronPasswordProvider provider = new WildFlyElytronPasswordProvider();
+
+    /**
+     * Produces a Modular Crypt Format bcrypt hash of the given password, using a generated salt and 10 iterations.
+     * 
+     * @param password the password to hash
+     * @return the Modular Crypt Format bcrypt hash of the given password
+     * @throws NullPointerException if the password is null
+     */
+    public static String bcryptHash(String password) {
+        return bcryptHash(password, 10);
+    }
+
+    /**
+     * Produces a Modular Crypt Format bcrypt hash of the given password, using a generated salt and the specified iteration
+     * count.
+     * 
+     * @param password the password to hash
+     * @param iterationCount the number of iterations to use while hashing
+     * @return the Modular Crypt Format bcrypt hash of the given password
+     * @throws NullPointerException if the password is null
+     * @throws IllegalArgumentException if the iterationCount parameter is negative or zero
+     */
+    public static String bcryptHash(String password, int iterationCount) {
+        byte[] salt = new byte[BCryptPassword.BCRYPT_SALT_SIZE];
+        random.nextBytes(salt);
+        return bcryptHash(password, iterationCount, salt);
+    }
+
+    /**
+     * Produces a Modular Crypt Format bcrypt hash of the given password, using the specified salt and the specified iteration
+     * count.
+     * 
+     * @param password the password to hash
+     * @param iterationCount the number of iterations to use while hashing
+     * @param salt the salt to use while hashing
+     * @return the Modular Crypt Format bcrypt hash of the given password
+     * @throws NullPointerException if the password or salt are null
+     * @throws IllegalArgumentException if the iterationCount parameter is negative or zero, or if the salt length is not equal
+     *         to 16
+     */
+    public static String bcryptHash(String password, int iterationCount, byte[] salt) {
+        if (iterationCount <= 0)
+            throw new IllegalArgumentException("Iteration count must be greater than zero");
+        Objects.requireNonNull(password, "password is required");
+        Objects.requireNonNull(salt, "salt is required");
+        if (salt.length != BCryptPassword.BCRYPT_SALT_SIZE)
+            throw new IllegalArgumentException("Salt lenght must be exactly " + BCryptPassword.BCRYPT_SALT_SIZE + " bytes");
```

Comment:
Here and also above, please use curly braces.

Reply:
If this is required, why doesn't the formatter do that, then?